File: src/content/fortran/bubblesort.f90
program bubblesort
    implicit none
    
    integer, parameter :: n = 10
    integer :: array(n)
    real :: realarray(n)
    integer :: i, j, temp
    
    ! Initialize the array with random integers
    call random_seed()
    call random_number(realarray)
    array = floor(realarray * 100)
    
    ! Print the original array
    print *, "Original array:"
    do i = 1, n
        write(*, '(I3)', advance='no') array(i)
    end do
    print *
    
    ! Bubble sort algorithm
    do i = 1, n-1
        do j = 1, n-i
            if (array(j) > array(j+1)) then
                temp = array(j)
                array(j) = array(j+1)
                array(j+1) = temp
            end if
        end do
    end do
    
    ! Print the sorted array
    print *, "Sorted array:"
    do i = 1, n
        write(*, '(I3)', advance='no') array(i)
    end do
    print *
    
end program bubblesort
    

File: src/content/pascal/bubblesort.pas
program BubbleSort;

const
    SIZE = 10;

var
    arr: array[1..SIZE] of Integer;
    i, j, temp: Integer;

begin
    Randomize;

    // Fill the array with random integers
    for i := 1 to SIZE do
        arr[i] := Random(100);

    // Print the original array
    Write('Original array: ');
    for i := 1 to SIZE do
        Write(arr[i], ' ');
    WriteLn;

    // Bubble sort algorithm
    for i := 1 to SIZE - 1 do begin
        for j := 1 to SIZE - i do begin
            if arr[j] > arr[j + 1] then begin
                temp := arr[j];
                arr[j] := arr[j + 1];
                arr[j + 1] := temp;
            end;
        end;
    end;

    // Print the sorted array
    Write('Sorted array: ');
    for i := 1 to SIZE do
        Write(arr[i], ' ');
    WriteLn;
end.